### PR TITLE
Enhancement/neural traces reordering

### PR DESCRIPTION
**Pull request by venus-sherathiya**
Let's go through neural traces reordering code tomorrow. I think it is working fine.

### CODE REVIEW

File: src/widgets/neuralWidget.py [hunk 1]
```diff
@@ -305,25 +310,25 @@ def loadNeural(self, ca_file, sample_rate, start_frame, stop_frame, time_start,
...
-        self.heatmap.setTransform(transform)
-        self.heatmap.setOpacity(0.5)
+        t_values = ((np.arange(self.data.shape[1]) - self.start_frame)/self.sample_rate) + self.time_start.float
+        y_values = np.arange(self.num_chans).reshape(-1,1) + 0.5 + self.normalize(self.data)
+        self.tracesOrder = y_values[:, self.start_frame]
+        self.drawTraces(t_values[self.start_frame:self.stop_frame], 
+                        y_values[:,self.start_frame:self.stop_frame])
+        #for chan in range(self.num_chans):
+        #    self.loadChannel(self.data, chan)
+        self.addTraces()
```
Comment (venus-sherathiya) [inline src/widgets/neuralWidget.py:320]: change "addTraces" to "putTracesIntoAGroup"

File: src/widgets/neuralWidget.py [hunk 2]
```diff
@@ -334,26 +339,65 @@ def loadNeural(self, ca_file, sample_rate, start_frame, stop_frame, time_start,
...
+            traceItem.setPen(pen)
+            self.traces.addToGroup(traceItem)
+
+    def reorderTracesAndHeatmap(self, showTraces, showHeatmap, showAnnotations):
+        for item in self.items():
+            if isinstance(item, QGraphicsItemGroup):
+                self.removeItem(item)
+            elif isinstance(item, QGraphicsPixmapItem):
+                self.removeItem(item)
+        newOrder = np.array([4,3,2,1,0])
```
Comment (venus-sherathiya) [inline src/widgets/neuralWidget.py:391]: remove this line

Comment (venus-sherathiya) [reply src/widgets/neuralWidget.py:391]: remove this line and make it an arguement

Comment (venus-sherathiya) [reply src/widgets/neuralWidget.py:391]: check newOrder size with current order size

File: src/widgets/neuralWidget.py [hunk 3]
```diff
@@ -333,26 +339,69 @@ def loadNeural(self, ca_file, sample_rate, start_frame, stop_frame, time_start,
...
+            elif isinstance(item, QGraphicsPixmapItem):
+                self.removeItem(item)
+        self.createHeatmap(newOrder)
+        offset = (self.tracesOrder[newOrder] - self.tracesOrder)
+        self.tracesOrder = self.tracesOrder + offset
+        for chan in list(self.tracesCache.keys()):
+            self.tracesCache[chan] = self.tracesCache[chan].translated(0., offset[int(chan)])
+        self.putTracesIntoAGroup()
+        self.addItem(self.traces)
+        self.setVisibility(showTraces, showHeatmap, showAnnotations)
```
Comment (annkennedy) [inline src/widgets/neuralWidget.py:403]: do we need to re-call setVisibility at the end of this function? Given that reorderTracesAndHeatmap is a function we would want plugin developers able to call, we should keep its number of required inputs as small as possible, either by setting default values of showTraces, showHeatmap, and showAnnotations, or just by not changing their values.

**Pull request merged**